static/src/javascripts/projects/common/modules/weather.js

```diff
@@ -0,0 +1,236 @@
+        },
+
+        getGeoLocation: function () {
+            // This sends all geolocation data to fetchData when fetchData is expecting an Accuweather location
+            navigator.geolocation.getCurrentPosition(this.fetchData, this.geoLocationDisabled);
+        },
+
+        geoLocationDisabled: function () {
+            self.changeLocationOptionText('error');
+        },
+
+        detectPosition: function (e) {
+            e.preventDefault();
+
+            self.changeLocationOptionText('process');
+            self.getGeoLocation();
+        },
+
+        changeLocationOptionText: function (state) {
+            $('.js-detect-location').text(getGeoStates[state]);
+        },
+
+        /**
+         * Check if user has data in local storage.
+         * If yes return data from local storage else return default location data.
+         *
+         * @returns {object} geolocation - lat and long
+         */
+        getUserLocation: function () {
+            var prefs = userPrefs.get(prefName);
+
+            if (prefs && prefs.id) {
+                return prefs;
+            }
+        },
+
+        getWeatherData: function (url) {
+            return ajax({
+                url: url,
+                type: 'json',
+                method: 'get',
+                crossOrigin: true
+            });
+        },
+
+        /**
+         * Save user location into localStorage
+         */
+        saveUserLocation: function (location) {
+            userPrefs.set(prefName, {
+                'id': location.id,
+                'city': location.city
+            });
+        },
+
+        getDefaultLocation: function () {
+            var location = this.getUserLocation();
+
+            if (location) {
+                this.fetchData(location);
+            } else {
+                try {
+                    self.getWeatherData(weatherApiUrl + '.json').then(function (response) {
+                        self.fetchData(response);
+                    });
+                } catch (e) {
+                    raven.captureException(new Error('Error retrieving city data (' + e.message + ')'), {
+                        tags: {
+                            feature: 'weather'
+                        }
+                    });
+                }
+            }
+        },
+
+        fetchData: function (location) {
+            self.saveUserLocation(location);
+
+            try {
+                return self.getWeatherData(weatherApiUrl + '/' + location.id + '.json').then(function (response) {
+                    self.render(response[0], location.city);
+                }).fail(function () {
+                    self.failedRequest();
+                });
+            } catch (e) {
+                raven.captureException(new Error('Error retrieving weather data (' + e.message + ')'), {
+                    tags: {
+                        feature: 'weather'
+                    }
+                });
+            }
```

Comment:
Why are you using `try/catch` here and the function above, instead of just relying on the `fail` promise?